clarity/src/vm/ast/definition_sorter/mod.rs

```diff
@@ -318,7 +324,11 @@ impl<'a> DefinitionSorter {
             .collect::<Vec<_>>();
 
         for pair in pairs.iter() {
-            self.probe_for_dependencies_in_key_value_pair(pair, tle_index, version)?;
+            self.probe_for_dependencies_in_key_value_pair(
+                pair.iter().collect(),
```

Comment:
why are you calling `pair.iter().collect()` here? did you mean to call `filter_comments`?

Reply:
😳 yes. Thanks. I'll add more tests to cover this too.

Reply:
Addressed in 06e546951.